coderd/database/dbauthz/dbauthz.go

```diff
@@ -3069,6 +3063,17 @@ func (q *querier) UpsertOAuthSigningKey(ctx context.Context, value string) error
 	return q.db.UpsertOAuthSigningKey(ctx, value)
 }
 
+func (q *querier) UpsertProvisionerDaemon(ctx context.Context, arg database.UpsertProvisionerDaemonParams) (database.ProvisionerDaemon, error) {
+	res := rbac.ResourceProvisionerDaemon
+	if arg.Tags[provisionersdk.TagScope] == provisionersdk.ScopeUser {
+		res = rbac.ResourceProvisionerDaemonUser.WithOwner(arg.Tags[provisionersdk.TagOwner])
+	}
```

Comment:
review: technically duplicated from `ProvisionerDaemon.RBACObject()` but not sure it's time yet to refactor this.